### PR TITLE
feat: weak supervision mode

### LINKED ISSUE #745: [Weak Supervision] Integrate labeling rules management from app

### PR DESCRIPTION
**Pull request by frascuchon**
Closes #745

### CODE REVIEW

File: frontend/components/commons/header/filters/FiltersArea.vue [hunk 1]
```diff
@@ -21,26 +21,36 @@
...
             @removeAllMetadataFilters="onRemoveAllMetadataFilters"
             @removeFiltersByGroup="onRemoveFiltersByGroup"
           ></FiltersList>
+          <re-button v-else-if="dataset.results.total" @click="showRulesList()" class="button-rules-summary button-clear">
+            <svgicon
+              name="config"
+              width="15"
+              height="14"
+            ></svgicon>
+            Rules Summary</re-button>
```
Comment (frascuchon) [inline frontend/components/commons/header/filters/FiltersArea.vue]: This piece of code is proper to the rules for text classification and should not be included in a common component in this way.

Some component extension system would have to be used for this functionality to be defined in the scope of the text classification task

File: frontend/components/commons/header/filters/FiltersArea.vue [hunk 2]
```diff
@@ -83,6 +98,14 @@ export default {
...
       });
     },
+    showRulesList() {
+      DatasetViewSettings.update({
+        where: this.dataset.name,
+        data: {
+          visibleRulesList: true,
+        },
+      });
+    },
```
Comment (frascuchon) [inline frontend/components/commons/header/filters/FiltersArea.vue]: ditto

File: frontend/components/commons/TaskSearch.vue [hunk 3]
```diff
@@ -20,7 +20,7 @@
     :class="[
       'app',
       currentTask,
-      annotationEnabled ? '--annotation' : '--exploration',
+      viewMode === 'annotate' ? '--annotation' : '--exploration',
```
Comment (frascuchon) [inline frontend/components/commons/TaskSearch.vue]: Could we just keep as it was defined `annotationEnabled` and rewrite how that property is computed in terms of new `viewSettings` structure ? 

For example:

```javascript
   computed: {
      annotationEnabled() {
         return this.viewMode === 'annotate';
      }
   }
```

File: frontend/components/commons/TaskSearch.vue [hunk 4]
```diff
@@ -33,7 +33,7 @@
         <task-sidebar :dataset="dataset" />
         <component :is="currentTaskHeader" :dataset="dataset" />
       </section>
-      <div :class="['grid', annotationEnabled ? 'grid--editable' : '']">
+      <div :class="['grid', viewMode === 'annotate' ? 'grid--editable' : '']">
```
Comment (frascuchon) [inline frontend/components/commons/TaskSearch.vue]: ditto

File: frontend/components/commons/header/filters/FiltersArea.vue [hunk 5]
```diff
@@ -21,26 +21,36 @@
       <div class="container">
         <div class="filters__row">
           <SearchBar
-            class="filters__searchbar"
+            :class="[viewMode === 'define-rules' ? '--extended' : null, 'filters__searchbar']"
```
Comment (frascuchon) [inline frontend/components/commons/header/filters/FiltersArea.vue]: Rules management is only for text classification. Should not be referenced here in that way

File: frontend/components/commons/header/filters/FiltersArea.vue [hunk 6]
```diff
@@ -164,4 +199,12 @@ export default {
...
   }
 }
+.button-rules-summary {
+  min-width: 130px !important;
+  color: $font-secondary-dark;
+  margin-bottom: 0 !important;
+  .svg-icon {
+    margin-right: 0.5em;
+  }
+}
```
Comment (frascuchon) [inline frontend/components/commons/header/filters/FiltersArea.vue]: Move into a n specific text classification component

File: frontend/components/commons/header/filters/SearchBar.vue [hunk 7]
```diff
@@ -99,22 +99,30 @@ export default {
...
+  &__button {
+    cursor: pointer;
+    padding: 5px;
+    border-radius: 3px;
+    background: $lighter-color;
+    transition: background 0.2s ease-in-out;
+    &:hover {
+      transition: background 0.2s ease-in-out;
+      background: palette(grey, bg)
+    }
```
Comment (frascuchon) [inline frontend/components/commons/header/filters/SearchBar.vue]: If these styles are related to the new rules management view shold not be in a common component.

If is just a style refactoring, should be in a separate PR. We have already a lot changes in this PR to review.

File: frontend/components/commons/header/filters/SearchBar.vue [hunk 8]
```diff
@@ -99,22 +99,30 @@ export default {
...
+    fill: #4C4EA3;
     margin: auto 1em auto 1em;
   }
   &:hover {
     box-shadow: 0px 3px 8px 3px rgba(222, 222, 222, 0.4)
   }
-  &__button {
-    cursor: pointer;
+  .--extended & {
+    min-width: 100%;
```
Comment (frascuchon) [inline frontend/components/commons/header/filters/SearchBar.vue:134]: ditto

File: frontend/components/commons/header/filters/SearchBar.vue [hunk 9]
```diff
@@ -99,22 +99,30 @@ export default {
...
   }
-  // &.re-input-focused {
-  //   border: 2px solid $secondary-color;
-  //   .svg-icon {
-  //     fill: $secondary-color;
-  //   }
-  // }
   .svg-icon {
-    fill: $primary-color;
+    fill: #4C4EA3;
```
Comment (frascuchon) [inline frontend/components/commons/header/filters/SearchBar.vue]: ditto

File: frontend/components/commons/header/global-actions/GlobalActions.vue [hunk 10]
```diff
@@ -16,7 +16,7 @@
   -->
 
 <template>
-  <div v-if="showAnnotationMode" class="container">
+  <div v-if="viewMode === 'annotate'" class="container">
```
Comment (frascuchon) [inline frontend/components/commons/header/global-actions/GlobalActions.vue]: rewrite computed property in terms of new `viewMode` attribute

File: frontend/components/commons/results/ResultsList.vue [hunk 11]
```diff
@@ -16,63 +16,69 @@
...
-      :pagination-settings="dataset.viewSettings.pagination"
-      @changePage="onPagination"
-    />
+      <RePagination
+        :total-items="dataset.results.total"
+        :pagination-settings="dataset.viewSettings.pagination"
+        @changePage="onPagination"
+      />
+    </template>
+    <rules-summary v-else :dataset="dataset"/>
```
Comment (frascuchon) [inline frontend/components/commons/results/ResultsList.vue]: common components must contain only common behaviour. Please, refactor and reuse common components in specific views

File: frontend/components/commons/results/ResultsList.vue [hunk 12]
```diff
@@ -118,6 +124,13 @@ export default {
       this.scrollComponent.addEventListener("scroll", this.onScroll);
     }
   },
+  updated() {
+    const scroll = document.getElementById("scroll");
+    if (scroll) {
+      this.scrollComponent = scroll;
+      this.scrollComponent.addEventListener("scroll", this.onScroll);
+    }
+  },
```
Comment (frascuchon) [inline frontend/components/commons/results/ResultsList.vue]: Is this related to rules summary  or just an old code fix? IF second case, please move this change to another PR

File: frontend/components/commons/results/ResultsRecord.vue [hunk 13]
```diff
@@ -19,26 +19,26 @@
   <div>
     <div
       :class="[
-        annotationEnabled ? 'list__item--annotation-mode' : 'list__item',
+        viewMode === 'annotate' ? 'list__item--annotation-mode' : 'list__item',
```
Comment (frascuchon) [inline frontend/components/commons/results/ResultsRecord.vue]: ditto

File: frontend/components/commons/results/ResultsRecord.vue [hunk 14]
```diff
@@ -19,26 +19,26 @@
...
         item.status === 'Discarded' ? 'discarded' : null,
       ]"
     >
       <ReCheckbox
-        v-if="annotationEnabled"
+        v-if="viewMode === 'annotate'"
```
Comment (frascuchon) [inline frontend/components/commons/results/ResultsRecord.vue]: ditto

File: frontend/components/commons/results/ResultsRecord.vue [hunk 15]
```diff
@@ -19,26 +19,26 @@
...
         class="list__checkbox"
         :value="item.selected"
         @change="onCheckboxChanged($event, item.id)"
       ></ReCheckbox>
       <slot :record="item" />
       <RecordExtraActions
         :key="item.id"
-        :allow-change-status="annotationEnabled"
+        :allow-change-status="viewMode === 'annotate'"
```
Comment (frascuchon) [inline frontend/components/commons/results/ResultsRecord.vue]: ditto

File: frontend/components/commons/results/ResultsRecord.vue [hunk 16]
```diff
@@ -19,26 +19,26 @@
...
         :record="item"
         :task="dataset.task"
         @onChangeRecordStatus="onChangeRecordStatus"
         @onShowMetadata="onShowMetadata(item)"
       />
-      <status-tag v-if="annotationEnabled && item.status !== 'Default'" :title="item.status"></status-tag>
+      <status-tag v-if="viewMode === 'annotate' && item.status !== 'Default'" :title="item.status"></status-tag>
```
Comment (frascuchon) [inline frontend/components/commons/results/ResultsRecord.vue]: ditto

File: frontend/components/commons/sidebar/SidebarMenu.vue [hunk 17]
```diff
@@ -35,15 +35,28 @@
         <a class="sidebar__info__button"
           href="#"
           data-title="Annotate"
-          @click="$emit('onEnableAnnotationView', true)"
+          @click="$emit('onChangeViewMode', 'annotate')"
         >
           <svgicon
-            v-show="annotationMode"
+            v-show="currentViewMode === 'annotate'"
```
Comment (frascuchon) [inline frontend/components/commons/sidebar/SidebarMenu.vue]: ditto

File: frontend/components/commons/sidebar/SidebarPanel.vue [hunk 18]
```diff
@@ -18,7 +18,7 @@
 <template>
   <aside
     :style="{ top: topPosition }"
-    :class="['sidebar', annotationEnabled ? 'annotation' : 'explore']"
+    :class="['sidebar', viewMode === 'annotate' ? 'annotation' : 'explore']"
```
Comment (frascuchon) [inline frontend/components/commons/sidebar/SidebarPanel.vue]: ditto

File: frontend/components/commons/sidebar/SidebarPanel.vue [hunk 19]
```diff
@@ -36,11 +36,11 @@ export default {
...
   },
   computed: {
-    annotationEnabled() {
-      return this.dataset.viewSettings.annotationEnabled;
+    viewMode() {
+      return this.dataset.viewSettings.viewMode;
     },
     topPosition() {
-      return this.annotationEnabled
+      return this.viewMode === 'annotate'
```
Comment (frascuchon) [inline frontend/components/commons/sidebar/SidebarPanel.vue]: ditto

File: frontend/components/commons/sidebar/SidebarMenu.vue [hunk 20]
```diff
@@ -35,15 +35,28 @@
...
+          data-title="Define rules"
+          @click="$emit('onChangeViewMode', 'define-rules')"
+        >
+          <svgicon
+            v-show="currentViewMode === 'define-rules'"
+            class="sidebar__info__icon-help"
+            name="check3"
+          ></svgicon>
+          <svgicon name="define-rules-view"></svgicon>
+        </a>
```
Comment (frascuchon) [inline frontend/components/commons/sidebar/SidebarMenu.vue]: could we find a mechanism to extends sidebar modes dinamically?

File: frontend/components/core/ReButton.vue [hunk 21]
```diff
@@ -221,6 +221,36 @@ export default {
...
+  }
+  &--small {
+    @extend .button-quaternary;
+    text-transform: none;
+    min-height: 30px;
+    line-height: 30px;
+    min-width: auto;
+  }
+}
+
```
Comment (frascuchon) [inline frontend/components/core/ReButton.vue]: Why the core button style is changed here?

File: frontend/components/core/ReSearchBar.vue [hunk 22]
```diff
@@ -60,12 +59,9 @@ export default {
   border-radius: 5px;
   @include font-size(14px);
   .svg-icon {
-    fill: $primary-color;
+    fill: $font-secondary;
```
Comment (frascuchon) [inline frontend/components/core/ReSearchBar.vue:68]: The core search bar changes also styles because the  rules management view ? This will affect to all uses of searchbar

File: frontend/components/text-classifier/results/RecordTextClassification.vue [hunk 23]
```diff
@@ -27,14 +27,14 @@
         :query-text="dataset.query.text"
       />
       <ClassifierAnnotationArea
-        v-if="annotationEnabled"
+        v-if="viewMode === 'annotate'"
```
Comment (frascuchon) [inline frontend/components/text-classifier/results/RecordTextClassification.vue]: ditto

File: frontend/components/text-classifier/results/RecordTextClassification.vue [hunk 24]
```diff
@@ -27,14 +27,14 @@
...
         :dataset="dataset"
         :record="record"
         @validate="validateLabels"
         @reset="resetLabels"
       />
       <ClassifierExplorationArea v-else :record="record" />
-      <div v-if="annotationEnabled" class="content__actions-buttons">
+      <div v-if="viewMode === 'annotate'" class="content__actions-buttons">
```
Comment (frascuchon) [inline frontend/components/text-classifier/results/RecordTextClassification.vue]: ditto

File: frontend/components/text-classifier/results/RecordTextClassification.vue [hunk 25]
```diff
@@ -44,7 +44,7 @@
       </div>
     </div>
 
-    <div v-if="!annotationEnabled" class="record__labels">
+    <div v-if="viewMode !== 'annotate'" class="record__labels">
```
Comment (frascuchon) [inline frontend/components/text-classifier/results/RecordTextClassification.vue]: ditto

File: frontend/components/text2text/results/RecordText2Text.vue [hunk 26]
```diff
@@ -29,7 +29,7 @@
           :record="record"
           :predictions="predictionSentences"
           :annotations="initialAnnotations"
-          :annotation-enabled="annotationEnabled"
+          :annotation-enabled="viewMode === 'annotate'"
```
Comment (frascuchon) [inline frontend/components/text2text/results/RecordText2Text.vue]: ditto

File: frontend/components/token-classifier/results/EntitiesHeader.vue [hunk 27]
```diff
@@ -29,7 +29,7 @@
       :class="[
         `color_${entity.colorId}`,
         activeEntity === entity.text ? 'active' : '',
-        annotationEnabled ? 'non-selectable--show-sort-code' : 'non-selectable',
+        viewMode === 'annotate' ? 'non-selectable--show-sort-code' : 'non-selectable',
```
Comment (frascuchon) [inline frontend/components/token-classifier/results/EntitiesHeader.vue]: ditto

File: frontend/components/token-classifier/results/EntitiesHeader.vue [hunk 28]
```diff
@@ -73,16 +73,16 @@ export default {
...
+      return this.dataset.viewSettings.viewMode;
     },
   },
   methods: {
     toggleEntitiesNumber() {
       this.showEntitySelector = !this.showEntitySelector;
     },
     onActiveEntity(entity) {
-      if (this.annotationEnabled) {
+      if (this.viewMode === 'annotate') {
```
Comment (frascuchon) [inline frontend/components/token-classifier/results/EntitiesHeader.vue]: ditto

File: frontend/components/token-classifier/results/EntityHighlight.vue [hunk 29]
```diff
@@ -27,13 +27,13 @@
       <span
         :class="[
           'highlight__tooltip',
-          annotationEnabled ? 'highlight__tooltip--icon' : '',
+          viewMode === 'annotate' ? 'highlight__tooltip--icon' : '',
```
Comment (frascuchon) [inline frontend/components/token-classifier/results/EntityHighlight.vue]: ditto

File: frontend/components/token-classifier/results/EntityHighlight.vue [hunk 30]
```diff
@@ -27,13 +27,13 @@
...
         ]"
       >
         <span
           >{{ span.entity.label }}
           <svgicon
-            v-if="annotationEnabled"
+            v-if="viewMode === 'annotate'"
```
Comment (frascuchon) [inline frontend/components/token-classifier/results/EntityHighlight.vue]: ditto

File: frontend/components/token-classifier/results/EntityHighlight.vue [hunk 31]
```diff
@@ -76,14 +76,14 @@ export default {
...
-      return this.dataset.viewSettings.annotationEnabled;
+    viewMode() {
+      return this.dataset.viewSettings.viewMode;
     },
   },
   methods: {
     openTagSelector() {
       this.clicked = true;
-      if (this.annotationEnabled) {
+      if (this.viewMode === 'annotate') {
```
Comment (frascuchon) [inline frontend/components/token-classifier/results/EntityHighlight.vue]: ditto

File: frontend/components/token-classifier/results/EntityHighlight.vue [hunk 32]
```diff
@@ -94,7 +94,7 @@ export default {
     },
     removeEntity() {
       this.doubleClicked = true;
-      if (this.annotationEnabled) {
+      if (this.viewMode === 'annotate') {
```
Comment (frascuchon) [inline frontend/components/token-classifier/results/EntityHighlight.vue]: ditto

File: frontend/components/token-classifier/results/RecordTokenClassification.vue [hunk 33]
```diff
@@ -34,7 +34,7 @@
         @reset="onReset"
       />
     </div>
-    <div v-if="annotationEnabled" class="content__actions-buttons">
+    <div v-if="viewMode === 'annotate'" class="content__actions-buttons">
```
Comment (frascuchon) [inline frontend/components/token-classifier/results/RecordTokenClassification.vue]: ditto

File: frontend/components/token-classifier/results/TextSpan.vue [hunk 34]
```diff
@@ -158,12 +158,12 @@ export default {
   },
   methods: {
     startSelection() {
-      if (this.annotationEnabled) {
+      if (this.viewMode === 'annotate') {
```
Comment (frascuchon) [inline frontend/components/token-classifier/results/TextSpan.vue]: ditto

File: frontend/database/modules/datasets.js [hunk 35]
```diff
@@ -597,6 +596,65 @@ const actions = {
...
+    );
+    return response.data;
+  },
+
+  async getRuleMetricsByLabel(_,{ dataset, query, label }) {
+    const { response } = await ObservationDataset.api().get(
+      `/datasets/${dataset.task}/${dataset.name}/labeling/rules/${query}/metrics?label=${label}`,
+    );
+    return response.data;
+  }
```
Comment (frascuchon) [inline frontend/database/modules/datasets.js]: All this actions should be defined only for text classification. 

Maybe better move  to dataset/modules/text_classification

**Comment by frascuchon** [comment]
Also, could you rename path `define-rules` to `labeling-rules` ?

File: frontend/components/commons/header/filters/SearchBar.vue (hunk 7 above)
Comment (leiyre) [reply frontend/components/commons/header/filters/SearchBar.vue]: This is a new style created for the rule search bar, but it affects other

File: frontend/components/commons/header/filters/SearchBar.vue (hunk 8 above)
Comment (leiyre) [reply frontend/components/commons/header/filters/SearchBar.vue:134]: this is a style that is used for rules but can be used elsewhere, it allows a larger version

File: frontend/components/core/ReButton.vue (hunk 21 above)
Comment (leiyre) [reply frontend/components/core/ReButton.vue]: it's a new button "button-quaternary"

File: frontend/components/core/ReSearchBar.vue (hunk 22 above)
Comment (leiyre) [reply frontend/components/core/ReSearchBar.vue:68]: the rules new design includes some little changes in commons components

File: frontend/components/commons/results/ResultsList.vue (hunk 12 above)
Comment (leiyre) [reply frontend/components/commons/results/ResultsList.vue]: this was to manage the header in the summary, but I'm going to change it

File: frontend/components/commons/results/ResultsList.vue [hunk 36]
```diff
@@ -16,32 +16,35 @@
...
       </div>
       <DynamicScroller
-        v-else
         page-mode
         class="scroller"
         :items="visibleRecords"
         :min-item-size="150"
         :buffer="200"
         :style="{ paddingTop: `${dataset.viewSettings.headerHeight + 10}px` }"
+        :emitUpdate="true"
```
Comment (frascuchon) [inline frontend/components/commons/results/ResultsList.vue]: Do we need this enable this ? Regarding documentation, could penalizes performance:

> emitUpdate (default: false): emit a 'update' event each time the virtual scroller content is updated (can impact performance).

File: frontend/components/text-classifier/labeling-rules/Rules.vue [hunk 37]
```diff
@@ -0,0 +1,70 @@
...
+  watch: {
+    async query(n, o) {
+      if (o !== n) {
+        await this.$fetch()
+      }
+    }
+  },
+  methods: {
+    updateRule() {
+      this.$fetch()
```
Comment (frascuchon) [inline frontend/components/text-classifier/labeling-rules/Rules.vue]: `await` should be included here

File: frontend/components/text-classifier/labeling-rules/Rules.vue [hunk 38]
```diff
@@ -0,0 +1,70 @@
...
+  },
+  watch: {
+    async query(n, o) {
+      if (o !== n) {
+        await this.$fetch()
+      }
+    }
+  },
+  methods: {
+    updateRule() {
```
Comment (frascuchon) [inline frontend/components/text-classifier/labeling-rules/Rules.vue]: this method should be defined as `async`

File: frontend/components/commons/results/ResultsEmpty.vue [hunk 39]
```diff
@@ -18,18 +18,28 @@
...
       default: "0 results found",
     },
+    emptyDescription: {
+      type: String,
+      default: undefined,
+    },
+    emptyIcon: {
+      type: String,
+      default: undefined,
+    },
```
Comment (frascuchon) [inline frontend/components/commons/results/ResultsEmpty.vue]: I think the naming is redundant since is called `ResultsEmpty`.

We can just parameterize as: `title`, `descrpition` and `icon`

File: frontend/components/commons/results/ResultsList.vue [hunk 40]
```diff
@@ -16,32 +16,34 @@
...
-        v-else
         page-mode
         class="scroller"
         :items="visibleRecords"
         :min-item-size="150"
         :buffer="200"
         :style="{ paddingTop: `${dataset.viewSettings.headerHeight + 10}px` }"
       > 
+        <template #before>
+          <slot name="pre-records-area"/>
```
Comment (frascuchon) [inline frontend/components/commons/results/ResultsList.vue]: Could we use the `results-header` name?

Comment (frascuchon) [reply frontend/components/commons/results/ResultsList.vue]: This "header" will be shown if no results was found?

File: frontend/components/commons/sidebar/SidebarMenu.vue [hunk 41]
```diff
@@ -35,15 +35,28 @@
...
+          data-title="Define rules"
+          @click="$emit('onChangeViewMode', 'labelling-rules')"
+        >
+          <svgicon
+            v-show="currentViewMode === 'labelling-rules'"
+            class="sidebar__info__icon-help"
+            name="check3"
+          ></svgicon>
+          <svgicon name="labelling-rules-view"></svgicon>
+        </a>
```
Comment (frascuchon) [inline frontend/components/commons/sidebar/SidebarMenu.vue]: We can encapsulate this as a `SidebarButton` component instead of repeat every time the same code.

File: frontend/components/core/ReSearchBar.vue [hunk 42]
```diff
@@ -26,14 +26,20 @@
...
 </template>
 <script>
 import "assets/icons/search";
 import "assets/icons/cross";
 
 export default {
+  props: {
+    placeholder: {
+      type: String,
+      required: true,
```
Comment (frascuchon) [inline frontend/components/core/ReSearchBar.vue]: We can define a generic value for placeholder: `Search`. It's too much require a placeholder value

File: frontend/components/core/ReTableInfo.vue [hunk 43]
```diff
@@ -47,150 +47,160 @@
...
-                    {{ itemValue(item, column) }}
+      <ResultsEmpty v-if="!data.length" :empty-title="emptyTitle" :empty-description="emptyDescription" :empty-icon="emptyIcon" />
+      <template v-else-if="resultsAvailable">
+        <div v-for="group in groups" :key="group" class="table-info__body">
+          <span v-if="groupBy && groupBy !== 'list'" class="table-info__group">
+            <p
+              class="table-info__group__title"
+              :class="{
+                model: groupBy === 'model',
+                datasource: groupBy === 'dataSource',
```
Comment (frascuchon) [inline frontend/components/core/ReTableInfo.vue]: a core component cannot contains datasource references...

File: frontend/components/core/ReTableInfo.vue [hunk 44]
```diff
@@ -47,150 +47,160 @@
...
+              }"
+            >
+              <svgicon
+                v-if="groupBy === 'dataSource'"
```
Comment (frascuchon) [inline frontend/components/core/ReTableInfo.vue]: ditto

File: frontend/components/core/ReTableInfo.vue [hunk 45]
```diff
@@ -47,150 +47,160 @@
...
+                        {{ itemValue(item, column)[key] }}
+                      </p>
+                    </span>
+                    <span v-else-if="column.type === 'task'">
+                      {{ itemValue(item, column) }}
+                      <span
+                        v-if="itemValue(item, column) === 'Text2Text'"
+                        class="table-info__tag"
+                        >Experimental</span
+                      >
```
Comment (frascuchon) [inline frontend/components/core/ReTableInfo.vue:114]: same as before: a core component cannot include references to task or another app logic

File: frontend/components/core/ReTableInfo.vue [hunk 46]
```diff
@@ -226,6 +236,22 @@ export default {
...
+      default: undefined,
+    },
+    emptyDescription: {
+      type: String,
+      default: undefined,
+    },
+    emptyIcon: {
+      type: String,
+      default: undefined,
+    },
```
Comment (frascuchon) [inline frontend/components/core/ReTableInfo.vue]: We can separate props in 2 blocks:

- Info when no data
- Info when component search has no results

````javascript
noDataInfo: {
   title: "",
   description: "", // maybe message is a better naming?
   icon: "",
},
emptySeachInfo: {
   title: "",
   description: "", // maybe message is a better naming?
   icon: "",
},
````

File: frontend/components/text-classifier/labeling-rules/Rules.vue [hunk 47]
```diff
@@ -0,0 +1,115 @@
+<template>
```
Comment (frascuchon) [inline frontend/components/text-classifier/labeling-rules/Rules.vue]: Rename to `RuleDefinition`

File: frontend/components/text-classifier/labeling-rules/RulesAnnotationArea.vue [hunk 48]
```diff
@@ -0,0 +1,280 @@
+<!--
```
Comment (frascuchon) [inline frontend/components/text-classifier/labeling-rules/RulesAnnotationArea.vue]: Rename to `RuleLabelsDefinition`

File: frontend/components/text-classifier/labeling-rules/RulesAnnotationArea.vue [hunk 49]
```diff
@@ -0,0 +1,280 @@
...
+  -
+  - Unless required by applicable law or agreed to in writing, software
+  - distributed under the License is distributed on an "AS IS" BASIS,
+  - WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  - See the License for the specific language governing permissions and
+  - limitations under the License.
+  -->
+
+<template>
+  <div class="annotation-area">
```
Comment (frascuchon) [inline frontend/components/text-classifier/labeling-rules/RulesAnnotationArea.vue]: Calling `annotation` can be confusing. It's not annotation, it's just linking labels to query

File: frontend/components/text-classifier/labeling-rules/Rules.vue [hunk 50]
```diff
@@ -0,0 +1,115 @@
...
+        class="rules__button button-secondary--outline"
+        @click="showRulesList"
+        >Manage rules</re-button
+      >
+      <input
+        v-model="description"
+        :class="[isFocused ? 'focused' : null, 'rules__query']"
+        @focus="isFocused = true"
+        @blur="isFocused = false"
+      />
```
Comment (frascuchon) [inline frontend/components/text-classifier/labeling-rules/Rules.vue]: Could we move this to `rules-annotation-area` component?

File: frontend/components/text-classifier/labeling-rules/RulesAnnotationArea.vue [hunk 51]
```diff
@@ -0,0 +1,280 @@
...
+    <div v-else class="empty-labels">
+      <p>This doesn't have any labels yet.</p>
+      <p>
+        To create new rules you need al least two labels. It's highly recommended to also annotate some records with these labels. Go to the annotation mode to 
+        <a href="#" @click.prevent="changeToAnnotationViewMode">create the labels and annotate some records</a>.
+      </p>
+    </div>
+    <rule-annotation-area-metrics :metrics="metrics" />
+    <p v-if="currentRule && selectedLabels.includes(currentRule.label) && currentRule.description === description">
+      {{saved ? 'Rule has been saved' : 'You have already a rule saved with that label'}}.
```
Comment (frascuchon) [inline frontend/components/text-classifier/labeling-rules/RulesAnnotationArea.vue]: Loading a rule from Management rules view raise message `You have already a rule saved with that label`. This should be avoided.

Maybe we can improve condition

File: frontend/pages/ws/_workspace/index.vue [hunk 52]
```diff
@@ -109,6 +110,12 @@ export default {
     workspace() {
       return currentWorkspace(this.$route);
     },
+    getDeleteModalContent() {
+      return {
+        title: 'Delete confirmation',
+        text: `You are about to delete: <strong>${this.showModal}</strong>. This action cannot be undone`
```
Comment (frascuchon) [inline frontend/pages/ws/_workspace/index.vue]: `showModal` is not a proper name for the rule name

File: frontend/components/commons/results/ResultsList.vue (hunk 40 above)
Comment (leiyre) [reply frontend/components/commons/results/ResultsList.vue]: If no results, this header is hidden

File: frontend/components/commons/results/Results.vue [hunk 53]
```diff
@@ -16,20 +16,29 @@
   -->
 
 <template>
-  <results-empty v-if="dataset.results.total === 0" />
+  <results-empty :title="emptySeachInfo.title" :message="emptySeachInfo.message" :icon="emptySeachInfo.icon" v-if="dataset.results.total === 0" />
```
Comment (frascuchon) [inline frontend/components/commons/results/Results.vue]:
```suggestion
  <results-empty :title="emptySearchInfo.title" :message="emptySearchInfo.message" :icon="emptySearchInfo.icon" v-if="dataset.results.total === 0" />
```

File: frontend/components/commons/results/Results.vue [hunk 54]
```diff
@@ -16,20 +16,29 @@
...
 export default {
   props: {
     dataset: {
       type: Object,
       default: () => ({}),
     },
   },
+  data: () => {
+    return {
+      emptySeachInfo: {
```
Comment (frascuchon) [inline frontend/components/commons/results/Results.vue]:
```suggestion
      emptySearchInfo: {
```

File: frontend/components/commons/results/ResultsList.vue [hunk 55]
```diff
@@ -16,32 +16,34 @@
   -->
 
 <template>
-  <div class="list">
+  <div class="content">  
     <slot name="header" />
```
Comment (frascuchon) [inline frontend/components/commons/results/ResultsList.vue:20]: I just saw it. Could we use this header and remove the `results-header` slot?

File: frontend/components/core/ReTableInfo.vue [hunk 56]
```diff
@@ -47,150 +47,150 @@
...
-                    </ReButton>
-                  </div>
+                  </re-action-tooltip>
                 </div>
-              </ReModal>
+                <ReModal
+                  :modal-custom="true"
+                  :prevent-body-scroll="true"
+                  modal-class="modal-secondary"
+                  :modal-visible="showModal === item.name"
```
Comment (frascuchon) [inline frontend/components/core/ReTableInfo.vue]: I've just realized that we don't set the `showModal` data property in any place. What's the logic about `showModal`?

File: frontend/components/text-classifier/labeling-rules/RulesSummary.vue [hunk 57]
```diff
@@ -0,0 +1,241 @@
+<template>
```
Comment (frascuchon) [inline frontend/components/text-classifier/labeling-rules/RulesSummary.vue]: Since we have no `summary` concept anymore. We can align naming and call the component as `RulesManagement` ?

File: frontend/components/text-classifier/results/TextClassificationResultsList.vue [hunk 58]
```diff
@@ -41,6 +46,11 @@ export default {
   data: () => ({
     selectedMetadataItem: undefined,
   }),
+  computed: {
+    showRulesArea() {
+      return !this.dataset.isMultiLabel && this.dataset.viewSettings.viewMode === 'labelling-rules';
```
Comment (frascuchon) [inline frontend/components/text-classifier/results/TextClassificationResultsList.vue]:
```suggestion
      return this.dataset.viewSettings.viewMode === 'labelling-rules';
```

File: frontend/components/commons/results/ResultsList.vue (hunk 55 above)
Comment (leiyre) [reply frontend/components/commons/results/ResultsList.vue:20]: Not really, the header is placed at the top outside the scrolling part

File: frontend/components/core/ReTableInfo.vue (hunk 56 above)
Comment (leiyre) [reply frontend/components/core/ReTableInfo.vue]: showModal is the id of the item whose modal is visible, I will rename it to visibleModalId

File: frontend/components/text-classifier/labeling-rules/RulesAnnotationArea.vue (hunk 51 above)
Comment (leiyre) [reply frontend/components/text-classifier/labeling-rules/RulesAnnotationArea.vue]: copy changed to "This query is already saved as a rule"

File: frontend/components/commons/results/ResultsList.vue [hunk 59]
```diff
@@ -16,32 +16,36 @@
   -->
 
 <template>
-  <div class="list">
-    <slot name="header" />
+  <div class="content">  
+    <slot name="header" />sssssdhuiehdiuehwudi
```
Comment (frascuchon) [inline frontend/components/commons/results/ResultsList.vue]:
```suggestion
    <slot name="header" />
```

File: frontend/components/commons/results/ResultsList.vue [hunk 60]
```diff
@@ -16,32 +16,36 @@
...
         class="scroller"
         :items="visibleRecords"
         :min-item-size="150"
         :buffer="200"
         :style="{ paddingTop: `${dataset.viewSettings.headerHeight + 10}px` }"
       > 
+        <template #before>
+          <slot name="results-header"/>
+          <results-empty :title="emptySearchInfo.title" :message="emptySearchInfo.message" :icon="emptySearchInfo.icon" v-if="dataset.results.total === 0" />
+        </template>
```
Comment (frascuchon) [inline frontend/components/commons/results/ResultsList.vue:36]: Why we cannot use the `header` slot to handle this info ?

Comment (leiyre) [reply frontend/components/commons/results/ResultsList.vue:36]: Because `header` slot  is outside the scroll area and we want it inside

File: .github/workflows/ci.yml [hunk 61]
```diff
@@ -94,7 +94,7 @@ jobs:
         with:
           context: .
           file: release.Dockerfile
-          push: ${{ github.event_name != 'pull_request' }}
+          # push: ${{ github.event_name != 'pull_request' }}
```
Comment (frascuchon) [inline .github/workflows/ci.yml]:
```suggestion
          push: ${{ github.event_name != 'pull_request' }}
```

File: frontend/components/commons/results/ResultsList.vue (hunk 60 above)
Comment (frascuchon) [reply frontend/components/commons/results/ResultsList.vue:36]: When we are using the `header` slot?
